test/builtin/specialfns/test_bessel.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for mathics.builtins.arithmetic.bessel
+"""
+from test.helper import check_evaluation
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    ("str_expr", "str_expected", "assert_failure_msg"),
+    # Basically special rules from autoload/rules/Bessel.m that are not covered
+    # by SymPy.
+    [
+        (
+            "BesselI[1/2,z]",
```

Comment:
I should note that the rule we have here for BesselI and BesselJ (from SymJa) apparently is not the most general rule in WMA. See for example: https://wolfram.com/xid/0e7q3bup-rrrwu

Reply:
Do you mean the spherical Bessel case?

Reply:
> Do you mean the spherical Bessel case?

I found what I was looking at before: https://wolfram.com/xid/0cq0os16-dlr12e

From BesselI Possible Issues:
> For symbolic arguments they [the BesselI arguments] are: 
> In[2]: Bessel[11/2, X] 

Note that we only handle +/- 1/2.

Reply:
And also under Specific values: 

> For half-integer orders, [BesselI](https://reference.wolfram.com/language/ref/BesselI.html) evaluates to elementary functions ...

Reply:
Yep, for half-integer indices, Bessel functions reduce to Spherical Bessel functions, which can be written in terms of  powers of e^{i r} and r. We can implement this later. By now, let's merge this as it is and then we can improve it.